app/controllers/provider_interface/candidates_controller.rb

```diff
@@ -0,0 +1,24 @@
+module ProviderInterface
+  class CandidatesController < ProviderInterfaceController
+    before_action :disable_on_production, only: :impersonate
+
+    def impersonate
+      candidate = Candidate.find(params[:candidate_id])
```

Comment:
Just to note that this allows providers to impersonate any candidate, not just candidates that have applications that they have access to.

Reply:
We've decided to add a simple guard in the controller checking there is an association between provider and candidate.